### DISCUSSION TITLE
Use local or external login success page?

### DISCUSSION THREAD

**Discussion by Blockyheadman** [GodotNuts/GodotFirebase#323] [Q&A] [answered]
I'm trying to make my Google OAuth2 redirect happen, but I have no idea how to do that with a local file. Is there a known way on how to use a local file? And if it's any better/ just because, I'd like to know how I can make it using a website file I have.

**Comment by fenix-hub** [comment, marked as answer]
To use a local file, OAuth2 must be configured with a redirect URL to localhost. This requires you to serve a static page on localhost. We do this by default, but instead of creating a specific file, for now the plugin automatically returns an HTML response directly to the browser. You can read that here
https://github.com/GodotNuts/GodotFirebase/blob/4df1a40144e9c1ecba85f6abf3021057bd33ec5a/addons/godot-firebase/auth/auth.gd#L343-L353

Anyway, OAuth2 is supposed to happen between online platforms, and localhost is generally used to test your flow.
It is preferred to use a redirect uri redirecting to your personal website/hosting with a static page.
Here you can read some more: https://github.com/GodotNuts/GodotFirebase/wiki/Authentication-and-User-Management#login-with-oauth-automatic
If you need help, please let us know

**Reply by Blockyheadman** [reply]
seems like i just had to update the firebase plugin to the latest github files and now it works fine. thanks for the help!
